src/components/study-container.js

```diff
@@ -0,0 +1,455 @@
+    studyUuid,
+    selectedNodeUuid,
+    fetcher,
+    invalidations,
+    defaultValue,
+    resultConversion
+) {
+    const [result, setResult] = useState(defaultValue);
+    const [isPending, setIsPending] = useState(false);
+    const [errorMessage, setErrorMessage] = useState(undefined);
+    const selectedNodeUuidRef = useRef();
+    const studyUpdatedForce = useSelector((state) => state.studyUpdated);
+    const lastUpdateRef = useRef();
+
+    const update = useCallback(() => {
+        selectedNodeUuidRef.current = selectedNodeUuid;
+        setIsPending(true);
+        fetcher(studyUuid, selectedNodeUuid)
+            .then((res) => {
+                if (selectedNodeUuidRef.current === selectedNodeUuid)
+                    setResult(resultConversion ? resultConversion(res) : res);
+            })
+            .catch((err) => setErrorMessage(err.message))
+            .finally(() => setIsPending(false));
+    }, [
+        selectedNodeUuid,
+        studyUuid,
+        setResult,
+        setErrorMessage,
+        setIsPending,
+        selectedNodeUuidRef,
+        fetcher,
+        resultConversion,
+    ]);
+
+    /* initial fetch and update */
+    useEffect(() => {
+        if (!studyUuid || !selectedNodeUuid) return;
+        const headers = studyUpdatedForce?.eventData?.headers;
+        const updateType = headers && headers['updateType'];
+        const node = headers && headers['node'];
+        const isUpdateForUs =
+            lastUpdateRef.current !== studyUpdatedForce &&
+            updateType &&
+            node === selectedNodeUuid &&
+            invalidations.find((e) => updateType === e) !== -1;
+        lastUpdateRef.current = studyUpdatedForce;
+        if (selectedNodeUuidRef.current !== selectedNodeUuid || isUpdateForUs) {
+            update();
+        }
+    }, [
+        update,
+        selectedNodeUuidRef,
+        selectedNodeUuid,
+        invalidations,
+        studyUpdatedForce,
+        studyUuid,
+    ]);
+
+    return [result, isPending, errorMessage, update];
+}
+
+function useStudy(studyUuidRequest) {
+    const [studyUuid, setStudyUuid] = useState(undefined);
+    const [pending, setPending] = useState(true);
+    const [errMessage, setErrMessage] = useState(undefined);
+    const intlRef = useIntlRef();
+
+    useEffect(() => {
+        fetchStudyExists(studyUuidRequest)
+            .then((response) => {
+                if (response.status === 200) {
+                    setStudyUuid(studyUuidRequest);
+                } else {
+                    setErrMessage(
+                        intlRef.current.formatMessage(
+                            { id: 'studyNotFound' },
+                            { studyUuid: studyUuidRequest }
+                        )
+                    );
+                }
+            })
+            .catch((e) => setErrMessage(e.message))
+            .finally(() => setPending(false));
+    }, [studyUuidRequest, setPending, setStudyUuid, intlRef]);
```

Comment:
set... are useless dependencies

Reply:
intlRef as well

Reply:
done

Reply:
intelRef is via a custom hook, had to reinsert it - react do not see it as a ref